src/main/java/org/terasology/metalrenegades/ai/system/CitizenSpawnSystem.java

```diff
@@ -124,6 +132,12 @@ private EntityRef spawnCitizen(EntityRef homeEntity) {
 
         EntityRef entityRef = entityBuilder.build();
 
+        if(entityRef.hasComponent(DisplayNameComponent.class)) {
+            DisplayNameComponent displayNameComponent = entityRef.getComponent(DisplayNameComponent.class);
+            CreatureNameProvider creatureNameProvider = new CreatureNameProvider(random.nextLong(), OLD_WEST);
+            displayNameComponent.name = creatureNameProvider.generateName();
```

Comment:
Do you also need to save the component again?

Reply:
Yea, will add that. However, I feel like it doesn't really make a difference here.